src/main/java/io/zksync/signer/ZkSigner.java

```diff
@@ -42,21 +47,49 @@ private ZkSigner(ZksPrivateKey privateKey, ZksCrypto crypto) {
     }
 
     public static ZkSigner fromSeed(byte[] seed) {
+        try {
+            // Generate private key from seed
+            ZksPrivateKey privateKey = crypto.generatePrivateKey(seed);
 
-        // Load native library
-        ZksCrypto crypto = ZksCrypto.load();
+            return new ZkSigner(privateKey);
+        } catch (ZksSeedTooShortException e) {
+            throw new ZkSyncException(e);
+        }
+    }
 
+    public static ZkSigner fromRawPrivateKey(byte[] rawPrivateKey){
         try {
-
             // Generate private key from seed
-            ZksPrivateKey privateKey = crypto.generatePrivateKey(seed);
+            ZksPrivateKey privateKey = crypto.generatePrivateKey(rawPrivateKey);
+            privateKey.data = rawPrivateKey;
 
-            return new ZkSigner(privateKey, crypto);
+            return new ZkSigner(privateKey);
         } catch (ZksSeedTooShortException e) {
             throw new ZkSyncException(e);
         }
     }
 
+    public static ZkSigner fromEthSigner(EthSigner ethSigner, ChainId chainId) {
+        String message = MESSAGE;
+        if (chainId != ChainId.Mainnet) {
+            message = String.format("%s\nChain ID: %d.", MESSAGE, chainId.getId());
+        }
+        EthSignature signature = ethSigner.signMessage(message, true);
+        if (signature.getType() != SignatureType.EthereumSignature) {
+            throw new ZkSyncIncorrectCredentialsException("Invalid signature type: " + signature.getType());
+        }
+        try {
+            if (!ethSigner.verifySignature(signature, message, true)) {
```

Comment:
What do you think about removing these checks? Here the user signs the message and then immediately verifies that it was indeed she who signed it. And that will be always true assuming that the crypto libraries are implemented correctly.